mass-update.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+# Define file directory and endings
+FILE_DIR="tmp/images"
+FILE_ENDINGS=".itb .bin"
+
+# Find files matching the specified endings
+FILES=""
+for ENDING in $FILE_ENDINGS; do
+    FILES="$FILES $(find "$FILE_DIR" -type f -name "*$ENDING")"
+done
+
+# Sort files based on whether filename contains "core" or not
+CORE_FILES=""
+OTHER_FILES=""
+for FILE_PATH in $FILES; do
+    if [[ "$FILE_PATH" == *"core"* ]]; then
+        CORE_FILES="$CORE_FILES $FILE_PATH"
+    else
+        OTHER_FILES="$OTHER_FILES $FILE_PATH"
+    fi
+done
+SORTED_FILES="$OTHER_FILES $CORE_FILES"
+
+# Print information and prompt for confirmation
+echo ""
+echo "This script will do the following:"
+echo ""
+echo "- flash all the following hosts with the corresponding firmware files currently present in $FILE_DIR"
+echo "- first flash APs, than core routers based on the naming convention"
+echo "- check the availability of the hosts before and after flashing"
+echo "- remove existing host keys and replace them with the current host keys"
+echo "- make sure that at least 16 MB of RAM are available before performing a sysupgrade"
+echo "- delete the local firmware file from disk after flashing"
+echo ""
+echo "The following files will be processed:"
+for FILE_PATH in $SORTED_FILES; do
+    echo "- $(basename "$FILE_PATH")"
+done
+echo ""
+read -rp "Do you want to proceed [y/N]? " choice
+echo ""
+if [[ ! "$choice" =~ ^[Yy]$ ]]; then
+    echo "Exiting..."
+    exit 0
+fi
+
+# Loop through each file
+for FILE_PATH in $SORTED_FILES; do
+    # Horizontal line to separate iterations
+    echo "----------------------------------------"
+
+    # Extract filename
+    FILENAME=$(basename "$FILE_PATH")
+    echo "Processing file: $FILENAME"
+
+    # Build nodename by omitting the ending
+    NODENAME="${FILENAME%.*}"
+    echo "Nodename: $NODENAME"
+
+    # Build hostname
+    HOSTNAME="$NODENAME.olsr"
+    echo "Hostname: $HOSTNAME"
+
+    # Check if hostname is reachable
+    echo "Checking if $HOSTNAME is reachable..."
+    if ping -c 1 "$HOSTNAME" >/dev/null 2>&1; then
+        echo "Hostname $HOSTNAME is reachable"
+
+        # Removing old host key
+        echo "Removing old host key for $HOSTNAME"
+        ssh-keygen -R "$HOSTNAME" >/dev/null 2>&1
```

Comment:
You can circumvent all the host key stuff: `sshnocheck='ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no'`